PlayerWidgetJobService.onHandleWork throws NPE

**App version:** 1.6.6 beta

**Android version**: 4.4 - 8.1

**Stacktrace/Logcat**: 
```
java.lang.RuntimeException: 
  at android.os.AsyncTask$3.done (AsyncTask.java:353)
  at java.util.concurrent.FutureTask.finishCompletion (FutureTask.java:383)
  at java.util.concurrent.FutureTask.setException (FutureTask.java:252)
  at java.util.concurrent.FutureTask.run (FutureTask.java:271)
  at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1162)
  at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:636)
  at java.lang.Thread.run (Thread.java:764)
Caused by: java.lang.NullPointerException: 
  at de.danoeh.antennapod.core.service.PlayerWidgetJobService.onHandleWork (PlayerWidgetJobService.java:5683)
  at android.support.v4.app.JobIntentService$CommandProcessor.doInBackground (JobIntentService.java:391)
  at android.support.v4.app.JobIntentService$CommandProcessor.doInBackground (JobIntentService.java:382)
  at android.os.AsyncTask$2.call (AsyncTask.java:333)
  at java.util.concurrent.FutureTask.run (FutureTask.java:266)
```

--- a/core/src/main/java/de/danoeh/antennapod/core/service/PlayerWidgetJobService.java
+++ b/core/src/main/java/de/danoeh/antennapod/core/service/PlayerWidgetJobService.java
@@ -41,8 +41,8 @@ public class PlayerWidgetJobService extends JobIntentService {
             return;
         }
 
-        if (PlaybackService.isRunning && playbackService == null) {
-            synchronized (waitForService) {
+        synchronized (waitForService) {
+            if (PlaybackService.isRunning && playbackService == null) {
                 bindService(new Intent(this, PlaybackService.class), mConnection, 0);
                 while (playbackService == null) {
                     try {

--- a/core/src/main/java/de/danoeh/antennapod/core/service/playback/PlaybackService.java
+++ b/core/src/main/java/de/danoeh/antennapod/core/service/playback/PlaybackService.java
@@ -1680,6 +1680,9 @@ public class PlaybackService extends MediaBrowserServiceCompat {
      * an invalid state.
      */
     public int getDuration() {
+        if (mediaPlayer == null) {
+            return INVALID_TIME;
+        }
         return mediaPlayer.getDuration();
     }
 
@@ -1688,6 +1691,9 @@ public class PlaybackService extends MediaBrowserServiceCompat {
      * is in an invalid state.
      */
     public int getCurrentPosition() {
+        if (mediaPlayer == null) {
+            return INVALID_TIME;
+        }
         return mediaPlayer.getPosition();
     }
 

